src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy

```diff
@@ -475,11 +478,17 @@ class AutoRerouteV2Spec extends HealthCheckSpecification {
         }
 
         and: "Flow is rerouted due to switchUp event"
-        //TODO(andriidovhan) specify flow history verification(it is not implemented yet) Reroute reason: switchUp event
         Wrappers.wait(WAIT_OFFSET / 2) {
-            assert northbound.getFlowHistory(firstFlow.flowId).findAll { it.action == "Flow rerouting" }.size() == 2
-
-            assert northbound.getFlowHistory(secondFlow.flowId).findAll { it.action == "Flow rerouting" }.size() == 5
+            def firstFlowHistory = northbound.getFlowHistory(firstFlow.flowId).findAll { it.action == "Flow rerouting" }
+            assert firstFlowHistory.size() == 2
+            assert firstFlowHistory.find {
+                it.details == "Reason: initiated by Reroute topology: Switch '$switchPair1.src.dpId' online"
+            }
+            def secondFlowHistory = northbound.getFlowHistory(secondFlow.flowId).findAll { it.action == "Flow rerouting" }
+            assert secondFlowHistory.findAll { it.action == "Flow rerouting" }.size() == 5
```

Comment:
I'd appreciate a brief comment about why we expect '5' here. This is not obvious